src/app/pages/ConsensusDashboardPage/SnapshotEpoch.tsx

```diff
@@ -14,13 +15,27 @@ const StyledBox = styled(Box)(({ theme }) => ({
   marginBottom: theme.spacing(3),
 }))
 
+// We need to get the previous epoch to compute end_height for the current one
+// This may not be precise during abnormal network conditions, but such conditions never happened so far
+const epochsLimit = 2
+
 export const SnapshotEpoch: FC<{ scope: SearchScope }> = ({ scope }) => {
   const { t } = useTranslation()
   const statusQuery = useGetStatus(scope.network)
   const blockHeight = statusQuery?.data?.data.latest_block
-  // TODO: provide epoch number when API is ready
-  const epoch = undefined
-  const percentageValue = undefined
+  const epochsQuery = useGetConsensusEpochs(scope.network, { limit: epochsLimit })
+  const epochs = epochsQuery.data?.data.epochs
+  const epoch = epochs && epochs[0].id
+  let percentageValue = undefined
+
+  if (epochs && epochs[1]?.end_height) {
+    const epochDiffHeight = epochs[1]?.end_height - epochs[1]?.start_height
+
+    if (blockHeight) {
+      const completedBlocksInCurrentEpoch = blockHeight - epochs[0]?.start_height
+      percentageValue = completedBlocksInCurrentEpoch / epochDiffHeight
+    }
+  }
```

Comment:
As this won't really change that often can we memoize it(Due to re-renders)?

And maybe multiply percentageValue in here? As opposed to inside prop passing(`value={percentageValue * 100}`).

Reply:
I wouldn't memoize. Seems very likely that it would hurt performance: caching and comparing memoization keys will take more CPU than three math operations. And it would hurt readability